janusgraph-inmemory/src/main/java/org/janusgraph/diskstorage/inmemory/BufferPage.java

```diff
@@ -111,24 +113,62 @@ private int getEntryEndOffset(final int index) {
         }
     }
 
+    /**
+     * This figures out how the "valuePosition" was stored (i.e. 1 byte or 4), reads it correctly,
+     * and leaves the byte buffer (if given) at the beginning of the key value
+     */
+    static int readValPos(byte[] rawData, int entryOffset, ByteBuffer entryBuffer)
+    {
```

Comment:
nitpick: don't put the curly brace on a new line, to be consistent with the convention used in the codebase.